src/utils/audiences/index.ts

```diff
@@ -0,0 +1,36 @@
+import { Audience, FeatureConfig, Filter, Target, UpdateTargetParams } from '../../api/schemas'
+
+export const replaceAudienceIdsInFeatureConfigs = (
+    featureConfigs: FeatureConfig[],
+    audiences: Audience[]
+) => {
+    return featureConfigs.map((featureConfig) => {
+        const newFeatureConfig = structuredClone(featureConfig) as typeof featureConfig
+        newFeatureConfig.targets = newFeatureConfig.targets.map((target) => {
+            return replaceAudienceIdsInTarget(target, audiences) as typeof target
+        })
+        return newFeatureConfig
+    })
+}
+
+export const replaceAudienceIdsInTarget = (target: Target | UpdateTargetParams, audiences: Audience[]) => {
+    const newTarget = structuredClone(target) as typeof target
+    const filters = newTarget.audience.filters.filters.map((filter) => {
+        return replaceAudienceIdInFilter(filter, audiences)
+    })
+    newTarget.audience.filters.filters = filters
+    return newTarget
+}
+
+export const replaceAudienceIdInFilter = (filter: Filter, audiences: Audience[]) => {
+    const newFilter = structuredClone(filter) as Filter
+    if (newFilter.type === 'audienceMatch') {
+        const audienceIds = newFilter._audiences!
+        const audienceNames = audienceIds.map((audienceId) => {
+            const audience = audiences.find((audience) => audience._id === audienceId)
```

Comment:
it may make sense to build a map of audience IDs to audience names so that we don't need to search the list for each filter, considering we're working with a list of all audiences for the project

Reply:
we could also save ourselves from iterating the targets and filters again here if we just pass the audiences as an optional parameter to the tree builder, then it can check the value in the map if it encounters an audience match filter